backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -487,28 +475,53 @@ private void addReleaseTitle(XWPFDocument document, LicenseInfoParsingResult par
-        XWPFRun copyrightTitleRun = document.createParagraph().createRun();
-        addFormattedText(copyrightTitleRun, "Copyrights", FONT_SIZE, true);
-        for (String copyright : getReleaseCopyrights(parsingResult)) {
-            XWPFParagraph copyPara = document.createParagraph();
-            copyPara.setSpacingAfter(0);
-            setText(copyPara.createRun(), copyright);
+    private void addCopyrights(XWPFDocument document, Collection<LicenseInfoParsingResult> projectLicenseInfoResults) {
+        for (LicenseInfoParsingResult parsingResult : projectLicenseInfoResults) {
+            if (parsingResult.getStatus() == LicenseInfoRequestStatus.SUCCESS) {
+                XWPFRun copyrightTitleRun = document.createParagraph().createRun();
+                addFormattedText(copyrightTitleRun, "Copyrights", FONT_SIZE, true);
+                for (String copyright : getReleaseCopyrights(parsingResult)) {
+                    XWPFParagraph copyPara = document.createParagraph();
+                    copyPara.setSpacingAfter(0);
+                    setText(copyPara.createRun(), copyright);
+                }
+            }
         }
+
     }
 
-    private void addLicenses(XWPFDocument document, LicenseInfoParsingResult parsingResult, boolean includeObligations) throws TException {
-        XWPFRun licensesTitleRun = document.createParagraph().createRun();
-        addNewLines(licensesTitleRun, 1);
-        addFormattedText(licensesTitleRun, "Licenses", FONT_SIZE, true);
-        for (String licenseName : getReleasesLicenses(parsingResult)) {
-            XWPFParagraph licensePara = document.createParagraph();
-            licensePara.setSpacingAfter(0);
-            addBookmarkHyperLink(licensePara, licenseName, licenseName);
-            if (includeObligations) {
-                addLicenseObligations(document, licenseName);
+    private void addLicenses(XWPFDocument document, Collection<LicenseInfoParsingResult> projectLicenseInfoResults, boolean includeObligations) throws TException {
+        for (LicenseInfoParsingResult parsingResult : projectLicenseInfoResults) {
+            addReleaseTitle(document, parsingResult);
+            if (parsingResult.getStatus() == LicenseInfoRequestStatus.SUCCESS) {
+                XWPFRun licensesTitleRun = document.createParagraph().createRun();
+                addNewLines(licensesTitleRun, 1);
+                addFormattedText(licensesTitleRun, "Licenses", FONT_SIZE, true);
+                Set<String> licenseNames = getReleasesLicenses(parsingResult).stream().collect(Collectors.toCollection(TreeSet::new));
+                int licenseCount = 1;
+                StringBuilder licenseNameWithCount = new StringBuilder();
+                for (String licenseName : licenseNames) {
+                    XWPFParagraph licensePara = document.createParagraph();
+                    licensePara.setSpacingAfter(0);
+                    licenseNameWithCount.append("(").append(licenseCount).append(") ");
```

Comment:
It would look good to make it a single line.

Reply:
@smrutis1 , I have restructured the code.